### PR TITLE
Remove setCode and unsafeNotInitializingSetCode

### PR DESCRIPTION
**Pull request by turbolent**
⚠️ Depends on #352

After the new contract deployment API was added in #352, remove the old API, i.e. the `setCode` and `unsafeNotInitializingSetCode` functions.

This is a big breaking change, but supporting both APIs is almost impossible.

### CODE REVIEW

**Review by janezpodhostnik — APPROVED** [review]
Looks good to me! 👍

**Review by joshuahannan — APPROVED** [review]
Documentation looks good to me! 🆗

File: languageserver/integration/commands.go [hunk 1]
```diff
@@ -280,45 +282,83 @@ RetryLoop:
...
+	tx := deployContractTransaction(i.activeAddress, name, code)
 
 	_, err := i.sendTransactionHelper(conn, i.activeAddress, tx)
 	return nil, err
 }
 
+const deployContractTemplate = `
+transaction(name: String, code: [UInt8]) {
+  prepare(signer: AuthAccount) {
+	if signer.contracts.get(name: name) == nil {
```
Comment (jordanschalm) [inline languageserver/integration/commands.go:331]:
```suggestion
    if signer.contracts.get(name: name) == nil {
```

File: languageserver/integration/commands.go [hunk 2]
```diff
@@ -280,45 +282,83 @@ RetryLoop:
...
+      signer.contracts.add(name: name, code: code)
+    } else {
+      signer.contracts.update__experimental(name: name, code: code)
+    }
+  }
+}
```
Comment (turbolent) [inline languageserver/integration/commands.go:337]: This template could probably be added to the Go SDK. cc @psiemens

File: runtime/deferral_test.go [hunk 3]
```diff
@@ -1192,7 +1274,24 @@ func TestRuntimeStorageDeferredResourceDictionaryValues_Insertion(t *testing.T)
 		getSigningAccounts: func() []Address {
 			return []Address{signer}
 		},
-		updateAccountCode: func(address Address, code []byte) (err error) {
+		resolveLocation: func(identifiers []Identifier, location Location) []ResolvedLocation {
```
Comment (jordanschalm) [inline runtime/deferral_test.go:1277]: This function could be factored out, it's repeated quite a few times in this file. 
```suggestion
		resolveLocation: ContractAddressLocationResolver(t),
```

```go
func ContractAddressLocationResolver(t *testing.T)  func([]Identifier, Location) []ResolvedLocation {
  return func(identifiers []Identifier, location Location) []ResolvedLocation {
			require.Len(t, identifiers, 1)
			require.IsType(t, AddressLocation{}, location)

			return []ResolvedLocation{
				{
					Location: AddressContractLocation{
						AddressLocation: location.(AddressLocation),
						Name:            identifiers[0].Identifier,
					},
					Identifiers: identifiers,
				},
			}
		}
}
```

**Review by jordanschalm — APPROVED** [review]

File: runtime/deferral_test.go (hunk 3 above)
Comment (turbolent) [reply runtime/deferral_test.go]: Refactored in ad1298df5a98ff3760a9c1b3ebab14fe7a697646

**Pull request merged**